### PR TITLE
master

### PR DESCRIPTION
**Pull request by malvitvik**
My pointer-set PR

### CODE REVIEW

File: src/pointer/set/CarManager.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+                    throw new IllegalArgumentException("Parameter '" + params[0] +
+                            "' isn't allowed. Allowed body types: " + Arrays.toString(BodyType.values()));
+                }
+
+                remover.remove(bodyType, Float.parseFloat(params[1]), Float.parseFloat(params[2]));
+                break;
+        }
+    }
+
+    private void searchByOneParameter(String parameter) throws NumberFormatException {
```
Comment (AlexU238) [inline src/pointer/set/CarManager.java]: WTF??? Why is this method name is like that? What parameter?

File: src/pointer/set/CarManager.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+        if (bodyType != null) {
+            search.find(bodyType);
+            return;
+        }
+
+        search.find(Float.parseFloat(parameter));
+    }
+
+    private void searchByTwoParameters(String color, String diameter) throws IllegalArgumentException {
```
Comment (AlexU238) [inline src/pointer/set/CarManager.java]: what is this name? I can only see from parameters what it does

File: src/pointer/set/CarManager.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+        if (c == null) {
+            throw new IllegalArgumentException("Color '" + color + "' isn't allowed. Allowed colors: "
+                    + Arrays.toString(Color.values()));
+        }
+
+        search.find(Float.parseFloat(diameter), c);
+    }
+
+    private void searchByTreeParameters(String tire, String minDiameter, String maxDiameter) throws IllegalArgumentException {
```
Comment (AlexU238) [inline src/pointer/set/CarManager.java]: name?

File: src/pointer/set/CarManager.java (hunk 1 above)
Comment (malvitvik) [reply src/pointer/set/CarManager.java]: yep :)

File: src/pointer/set/CarManager.java (hunk 2 above)
Comment (malvitvik) [reply src/pointer/set/CarManager.java]: it is private

File: src/pointer/set/CarManager.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+            case 3:
+                searchByTreeParameters(params[0], params[1], params[2]);
+                break;
+            default:
+                System.out.println("Wrong parameters count.");
+                break;
+        }
+    }
+
+    void update(String[] params) throws NumberFormatException {
```
Comment (AlexU238) [inline src/pointer/set/CarManager.java]: what this method updates?

File: src/pointer/set/CarManager.java (hunk 3 above)
Comment (malvitvik) [reply src/pointer/set/CarManager.java]: private method

**Pull request closed without merging**